evm/src/memory/memory_stark.rs

```diff
@@ -0,0 +1,580 @@
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+use itertools::{izip, multiunzip, Itertools};
+use plonky2::field::extension_field::{Extendable, FieldExtension};
+use plonky2::field::packed_field::PackedField;
+use plonky2::field::polynomial::PolynomialValues;
+use plonky2::hash::hash_types::RichField;
+use plonky2::timed;
+use plonky2::util::timing::TimingTree;
+use rand::Rng;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::memory::registers::{
+    sorted_value_limb, value_limb, ADDR_CONTEXT, ADDR_SEGMENT, ADDR_VIRTUAL, CONTEXT_FIRST_CHANGE,
+    COUNTER, COUNTER_PERMUTED, IS_READ, NUM_REGISTERS, RANGE_CHECK, RANGE_CHECK_PERMUTED,
+    SEGMENT_FIRST_CHANGE, SORTED_ADDR_CONTEXT, SORTED_ADDR_SEGMENT, SORTED_ADDR_VIRTUAL,
+    SORTED_IS_READ, SORTED_TIMESTAMP, TIMESTAMP, VIRTUAL_FIRST_CHANGE,
+};
+use crate::permutation::PermutationPair;
+use crate::stark::Stark;
+use crate::util::{permuted_cols, trace_rows_to_poly_values};
+use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
+
+pub(crate) const NUM_PUBLIC_INPUTS: usize = 0;
+
+#[derive(Copy, Clone)]
+pub struct MemoryStark<F, const D: usize> {
+    pub(crate) f: PhantomData<F>,
+}
+
+pub struct MemoryOp<F> {
```

Comment:
I think it would be useful to also include the channel index in here, so that the generation code here can handle populating the `is_memop` flags.

Or if it's hard to change with the way the test is written, we can add a TODO for later. I think `generate_random_memory_ops` etc. can go away later when we have more "real" tests.